extensions/machine-learning-services/src/prediction/predictService.ts

```diff
@@ -0,0 +1,202 @@
+	 */
+	public async generatePredictScript(
+		predictParams: PredictParameters,
+		registeredModel: RegisteredModel
+	): Promise<string> {
+		let connection = await this.getCurrentConnection();
+		let query = '';
+		if (registeredModel.id) {
+			query = this.getPredictScriptWithModelId(
+				registeredModel.id || 0,
+				predictParams.inputColumns || [],
+				predictParams.outputColumns || [],
+				predictParams);
+		} else {
+			let modelBytes = await utils.readFileInHex(registeredModel.filePath || '');
+			query = this.getPredictScriptWithModelBytes(modelBytes, predictParams.inputColumns || [],
+				predictParams.outputColumns || [],
+				predictParams);
+		}
+		let document = await this._apiWrapper.openTextDocument({
+			language: 'sql',
+			content: query
+		});
+		await this._apiWrapper.showTextDocument(document.uri);
+		await this._apiWrapper.connect(document.uri.toString(), connection.connectionId);
+		this._apiWrapper.runQuery(document.uri.toString(), undefined, false);
+		return query;
+	}
+
+	/**
+	 * Returns list of tables given database name
+	 * @param databaseName database name
+	 */
+	public async getTableList(databaseName: string): Promise<DatabaseTable[]> {
+		let connection = await this.getCurrentConnection();
+		let list: DatabaseTable[] = [];
+		if (connection) {
+			let query = utils.withDbChange(connection.databaseName, databaseName, this.getTablesScript(databaseName));
+			let result = await this._queryRunner.safeRunQuery(connection, query);
+			if (result && result.rows && result.rows.length > 0) {
+				result.rows.forEach(row => {
+					list.push({
+						databaseName: databaseName,
+						tableName: row[0].displayValue,
+						schema: row[1].displayValue
+					});
+				});
+			}
+		}
+		return list;
+	}
+
+	/**
+	 *Returns list of column names of a database
+	 * @param databaseTable table info
+	 */
+	public async getTableColumnsList(databaseTable: DatabaseTable): Promise<string[]> {
+		let connection = await this.getCurrentConnection();
+		let list: string[] = [];
+		if (connection && databaseTable.databaseName) {
+			const query = utils.withDbChange(connection.databaseName, databaseTable.databaseName, this.getTableColumnsScript(databaseTable));
+			let result = await this._queryRunner.safeRunQuery(connection, query);
+			if (result && result.rows && result.rows.length > 0) {
+				result.rows.forEach(row => {
+					list.push(row[0].displayValue);
+				});
+			}
+		}
+		return list;
+	}
+
+	private async getCurrentConnection(): Promise<azdata.connection.ConnectionProfile> {
+		return await this._apiWrapper.getCurrentConnection();
+	}
+
+	private getTableColumnsScript(databaseTable: DatabaseTable): string {
+		return `
+SELECT COLUMN_NAME,*
+FROM INFORMATION_SCHEMA.COLUMNS
+WHERE TABLE_NAME='${utils.doubleEscapeSingleQuotes(databaseTable.tableName)}'
+AND TABLE_SCHEMA='${utils.doubleEscapeSingleQuotes(databaseTable.schema)}'
+AND TABLE_CATALOG='${utils.doubleEscapeSingleQuotes(databaseTable.databaseName)}'
```

Comment:
You would probably want a UT to check all single double quote and square bracket combinations work here...

Reply:
yes, will add unit test in coming PRs

---
In reply to: [389909543](https://github.com/microsoft/azuredatastudio/pull/9450#discussion_r389909543) [](ancestors = 389909543)